src/send_kcidb.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022 Maryam Yusuf
+# Author: Maryam Yusuf <maryam.m.yusuf1802@gmail.com>
+import datetime
+import os
+import sys
+
+import kernelci
+import kernelci.data
+from kernelci.config import load
+from kernelci.cli import Args, Command, parse_opts
+from kcidb import Client
+import kcidb
+
+
+class cmd_run(Command):
+    help = "Listen for events and send them to KCDIB"
+    args = [Args.db_config]
+    project_id = os.getenv('KCIDB_PROJECT_ID')
+    topic_name = os.getenv('KCIDB_TOPIC_NAME')
+    google_credential = os.getenv('GOOGLE_APPLICATION_CREDENTIALS')
+    client = None
+
+    if topic_name and project_id:
+        client = Client(project_id=project_id,
+                        topic_name=topic_name)
+
+    def __call__(self, configs, args):
+        db_config = configs['db_configs'][args.db_config]
+        api_token = os.getenv('API_TOKEN')
+        db = kernelci.data.get_db(db_config, api_token)
+
+        sub_id = db.subscribe('node')
+        print("Listening for events... ")
+        print("Press Ctrl-C to stop.")
+        sys.stdout.flush()
+        if not self.client or not self.google_credential:
+            print("Aborting due to missing configuration")
+            sys.stdout.flush()
+            return
+
+        try:
+            while True:
+                event = db.get_event(sub_id)
+                node = db.get_node_from_event(event)
+                if node['name'] != 'checkout':
+                    continue
+                print(f"Printing node for {node}")
+                sys.stdout.flush()
+                print("Submitting node to KCIDB")
+                sys.stdout.flush()
+                self.send_revision(node)
+
+            sys.stdout.flush()
+
+        except KeyboardInterrupt:
+            print("Stopping.")
+        finally:
+            db.unsubscribe(sub_id)
+
+        sys.stdout.flush()
+
+    def get_revision(self, node):
+        dt = datetime.datetime.fromisoformat(node["created"],)
+        tz_utc = datetime.timezone(datetime.timedelta(hours=0))
```

Comment:
This doesn't need to be recreated each time, it could be an attribute in the class set in `__init__` or `__call__`.